### PR TITLE
Aligned the /resources/personnel endpoint so that both apiversions uses same code

### PR DESCRIPTION
**Pull request by oivheg**
- [ ] New feature
- [x] Bug fix
- [ ] High impact

**Description of work:**
The endpoint   [HttpGet("departments/{fullDepartmentString}/resources/personnel")]  versions have been aligned to return the same information. This is done because the current version 1 used in production does not get the correct personell for the deparment du to the way it works. 

- Endpoint have been aligned to prevent the need of updating exsisting frontedn code
- removed Endpoint V2
- Aligning the endpoints results in the swagger page not showing the endpoint  for some reason. Testing shows that it returns.

[AB#41492](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/41492)

**Testing:**
- [x] Can be tested
- [ ] ~~Automatic tests created / updated~~
- [x] Local tests are passing

Can be tested by trying the endpoint with and wihtout api-version set and try 1 and 2. 
Both sould return the same data and return users that have the same department as specified in the endpoint. 

Testing against produciton gives different personell from other department when using V1, while v2 gives correct. 

Replace :department with  PDP PRD PMC QRI IDM2

**PR**
https://resources-api-pr-577.fusion-dev.net/departments/:deparment/resources/personnel?includeSubdepartments=false&timelineStart=2023-06-01&timelineEnd=2023-11-30&$expand=timeline&api-version=2

**CI**
https://resources-api-ci.fusion-dev.net/departments/:deparment/resources/personnel?includeSubdepartments=false&timelineStart=2023-06-01&timelineEnd=2023-11-30&$expand=timeline&api-version=2
**PROD**
https://fap-resources-api-fprd.azurewebsites.net/departments/:deparment/resources/personnel?includeSubdepartments=false&timelineStart=2023-06-01&timelineEnd=2023-11-30&$expand=timeline&api-version=1

**Checklist:**
- [x] Considered automated tests
- [ ] ~~Considered updating specification / documentation~~
- [ ] ~~Considered work items~~ 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs [hunk 1]
```diff
@@ -122,18 +125,19 @@ public InternalPersonnelController()
         [MapToApiVersion("2.0")]
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:125]: Could we not just do

```
[MapToApiVersion("1.0")]
[MapToApiVersion("2.0")]
public async Task<ActionResult<ApiCollection<ApiInternalPersonnelPerson>>> GetDepartmentPersonnel(string fullDepartmentString,
            [FromQuery] ODataQueryParams query,
Expand All
	@@ -46,7 +49,7 @@ public InternalPersonnelController()
            [FromQuery] DateTime? timelineEnd = null,
            [FromQuery] bool includeSubdepartments = false,
            [FromQuery] bool includeCurrentAllocations = false)
{
    var command = new GetDepartmentPersonnel(fullDepartmentString, query)
                .IncludeSubdepartments(includeSubdepartments)
                .IncludeCurrentAllocations(includeCurrentAllocations)
                .WithTimeline(shouldExpandTimeline, timelineStart, timelineEnd)
                .WithVersion(2); // No longer use v1
}
```

Could also clean up by just remove v1 logic in the query, so it is not mistaken agan....

Comment (oivheg) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:123]: i tough so to,  but there is something wierd happening making functions dissapear when i do that

Comment (oivheg) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:122]: when combining them the endpoint looks like this:  this should be on top departments/{fullDepartmentString}/resources/personnel
image

Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs]: Yeh, i know, but i think we can update the swagger config to find it.. They can be reached, they just don't show up in swagger

Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs]: Have updated the swagger logic to handle these endpoints now

Comment (oivheg) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs]: Now it looks fine, appears to be working as expected.

File: src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs [hunk 2]
```diff
@@ -203,7 +194,7 @@ public InternalPersonnelController()
 
             return new ApiCollection<ApiInternalPersonnelPerson>(returnModel);
         }
-
+        [MapToApiVersion("1.0")]
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:197]: Why is this needed?

File: src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs [hunk 3]
```diff
@@ -247,7 +238,7 @@ public async Task<ActionResult<ApiInternalPersonnelPerson>> GetPersonnelAllocati
 
             return Ok(result);
         }
-
+        [MapToApiVersion("1.0")]
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:241]: See above comment..

File: src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs [hunk 4]
```diff
@@ -13,10 +13,12 @@
 
 namespace Fusion.Resources.Api.Controllers
 {
-    [ApiVersion("1.0-preview")]
-    [ApiVersion("1.0")]
+
     [Authorize]
     [ApiController]
+    [ApiVersion("2.0")]
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:19]: We only version endpoints, not whole controllers..

**Review by HansDahle — CHANGES_REQUESTED** [review]

File: src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs (hunk 2 above)
Comment (oivheg) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:197]: I Tried removing, but then a lot of swaggers dissapear. 
I have also compared pr vs CI environemnt and there is a lot of differences 
image

Comment (oivheg) [reply src/backend/api/Fusion.Resources.Api/Controllers/Personnel/InternalPersonnelController.cs:197]: It appeasr the ones getting Obsolete is correct, but i am not able to not use     [MapToApiVersion("1.0")] on the InternalPerosnell endpoints that are supposed to b V1

**Review by HansDahle — APPROVED** [review]
